[16.0][MIG] sale_order_qty_change_no_recompute: Migration to version 16.0

Yadier-Tecnativa opened the pull request:
@Tecnativa TT42279

victoralmau committed:
[ADD] sale_order_qty_change_no_recompute: Prevent recompute if only quantity has changed in sale order line

pedrobaeza committed:
[FIX] sale_order_qty_change_no_recompute: Proper way for removing onchange

The _origin trick is not valid, as it doesn't reflect the state before
the onchange, only the saved copy (if any).

victoralmau committed:
[IMP] sale_order_qty_change_no_recompute: black, isort, prettier

victoralmau committed:
[MIG] sale_order_qty_change_no_recompute: Migration to 14.0

victoralmau committed:
[IMP] sale_order_qty_change_no_recompute: Improve tests

victoralmau committed:
[IMP] sale_order_qty_change_no_recompute: Add context in test to prevent onchange quantity and some possible related error in another addons.

pedrobaeza committed:
[IMP] sale_order_qty_change_no_recompute: Block also uom onchange

Changing that field shouldn't trigger price recomputation as well.

TT32663

cesar-tecnativa committed:
[MIG] sale_order_qty_change_no_recompute: Migration to 15.0

pedrobaeza commented on sale_order_qty_change_no_recompute/models/sale_order.py:
```diff
@@ -2,28 +2,34 @@
-        return super()._onchange_eval(field_name, onchange, result)
+    @api.depends("product_id", "product_uom", "product_uom_qty")
+    def _compute_price_unit(self):
+        for line in self:
```
You don't have to copy the upstream method. This should be restricted to just store the current price and discount before calling super, and then restore it after the call.

Yadier-Tecnativa replied:
The computed method had to be overwritten for when the dependent fields change and obtain a referential value to the old product, in order to compare, since the calculation of the new price depends on the product

pedrobaeza replied:
But we don't want that... this makes this module incompatible with any change in the price unit computations by other modules, as it's totally overwritten the way it's computed. We just want to not perform the computations on quantity changed. You have to find another technique.

pedrobaeza reviewed (changes requested):
I understand that the previous code is not working. Isn't it?

rousseldenis commented:
/ocabot migration sale_order_qty_change_no_recompute

pedrobaeza committed:
[MIG] sale_order_qty_change_no_recompute: Migration to 16.0

- Standard procedure
- Change the strategy, as now the change is done through a computed
  field.
- Improved README

pedrobaeza reviewed (approved):
Changed to a new approach to get the feature covered.

sergio-teruel reviewed (approved):
Code and functional review.
:+1:

pedrobaeza commented:
/ocabot merge nobump

sergio-teruel commented:
ping @juangomezzurita
Done!
